flake8: Undefined name 'source_file' in test_parse_manifest.py

### Describe the bug
A clear and concise description of what the bug is. What command did you run? What happened?

$ `flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics`
```
./dbt/test/unit/test_parse_manifest.py:105:71: F821 undefined name 'source_file'
        self.parser.parse_file.assert_called_once_with(FileBlock(file=source_file))
                                                                      ^
1     F821 undefined name 'source_file'
1
```
### Steps To Reproduce
In as much detail as possible, please provide steps to reproduce the issue. Sample data that triggers the issue, example model code, etc is all very helpful here.

$ `flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics`

### Expected behavior
A clear and concise description of what you expected to happen.

### Screenshots and log output
If applicable, add screenshots or log output to help explain your problem.

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)


**The output of `dbt --version`:**
```
<output goes here>
```

**The operating system you're using:**

**The output of `python --version`:**

### Additional context
Add any other context about the problem here.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -18,6 +18,9 @@
 ###  Docs
 - Fix missing data on exposures in docs ([#4467](https://github.com/dbt-labs/dbt-core/issues/4467))
 
+### Under the hood
+- Testing cleanup ([#4496](https://github.com/dbt-labs/dbt-core/pull/4496))
+
 Contributors:
 - [remoyson](https://github.com/remoyson) ([#4442](https://github.com/dbt-labs/dbt-core/pull/4442))
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,9 @@
 ### Features
 - New Dockerfile to support specific db adapters and platforms.  See docker/README.md for details ([#4495](https://github.com/dbt-labs/dbt-core/issues/4495), [#4487](https://github.com/dbt-labs/dbt-core/pull/4487))
 
+### Under the hood
+- Testing cleanup ([#4496](https://github.com/dbt-labs/dbt-core/pull/4496))
+
 ## dbt-core 1.0.1rc1 (December 20, 2021)
 
 ### Fixes
@@ -17,9 +20,6 @@
 
 ###  Docs
 - Fix missing data on exposures in docs ([#4467](https://github.com/dbt-labs/dbt-core/issues/4467))
-
-### Under the hood
-- Testing cleanup ([#4496](https://github.com/dbt-labs/dbt-core/pull/4496))
 
 Contributors:
 - [remoyson](https://github.com/remoyson) ([#4442](https://github.com/dbt-labs/dbt-core/pull/4442))

--- a/test/unit/test_parse_manifest.py
+++ b/test/unit/test_parse_manifest.py
@@ -101,8 +101,3 @@ class TestLoader(unittest.TestCase):
             project_root=normalize(self.root_project_config.project_root),
         )
         return SourceFile(path=path, checksum=checksum)
-
-        self.parser.parse_file.assert_called_once_with(FileBlock(file=source_file))
-
-# Note: none of the tests in this test case made sense with the removal
-# of the old method of partial parsing. 

